java-checks/src/test/files/checks/ImmediateReverseBoxingCheck.java

```diff
@@ -1,16 +1,18 @@
 abstract class A {
-  
+
   public void processInt(String s, int a) {}
   public void processInteger(String s, Integer... a) {}
-  
-  void intBoxingAndUnboxing(int int1, Integer integer1, String string) {
+
+  void intBoxingAndUnboxing(int int1, Integer integer1, String string, Double double1) {
```

Comment:
A case I don't see and I am wondering about covering it or not : 
```
Integer integer1 = 1;
new Integer(integer1); // shall raise an issue or not ? 
```
I think issue should be legit most of the time the only doubt I can think of is if you want to have two instances of the same value (but that seems a corner case). If you think we should cover that one, RSPEC might require a minor update.

Reply:
I would prefer not to raise an issue this. Like you said, if you want another instance with the same value, it would be the only way to have it without raising an issue, because `new Integer(integer1.intValue());` is already raising an issue related to unboxing (line 27 from test file).

Reply:
But this huge case seems highly arguable whereas detecting such issue when it is not intended will be valuable.

Reply:
Okay, so let's handle it. Adding the case to RSPEC and updating code.